troposphere/static/js/components/modals/instance/InstanceResizeModal.jsx

```diff
@@ -0,0 +1,268 @@
+import React from "react";
+import BootstrapModalMixin from "components/mixins/BootstrapModalMixin";
+import stores from "stores";
+import ResourcesForm from "components/modals/instance/launch/components/ResourcesForm";
+import globals from "globals";
+
+export default React.createClass({
+    displayName: "InstanceResizeModal",
+
+    mixins: [BootstrapModalMixin],
+
+    //
+    // Internal Modal Callbacks
+    // ------------------------
+    //
+
+    cancel: function() {
+        this.hide();
+    },
+
+    confirm: function(resize_size) {
+        this.hide();
+        this.props.onConfirm(resize_size);
+    },
+    confirmResize: function() {
+        this.hide();
+        let size = this.state.providerSize;
+        this.props.onConfirm(size.attributes.alias);
+    },
+
+    // Render
+    // ------
+    //
+
+    propTypes: {
+        project: React.PropTypes.instanceOf(Backbone.Model),
+        onConfirm: React.PropTypes.func.isRequired,
+    },
+
+    getInitialState: function() {
+
+        let project = this.props.project ? this.props.project : null;
+        // Check if the user has any projects, if not then set view to "PROJECT_VIEW"
+        // to create a new one
+        let projectList = stores.ProjectStore.getAll();
+
+        return {
+            provider: null,
+            // State for launch
+            project,
+            providerSize: null,
+            identityProvider: null,
+            allocationSource: null
+        }
+    },
+
+    // Set the component's state based on cloud defaults.
+    //
+    // Whenever the wizard mounts it listens for changes from the stores,
+    // passing this function as a callback. Incrementally it calls stores to
+    // fetch data. It only sets state for defaults, i.e. if project is null,
+    // set the project to the first returned from the cloud. It primes our
+    // stores, so that render can just call get and eventually get data.
+    updateState: function() {
+        let allocationSourceList = stores.AllocationSourceStore.getAll();
+
+
+        // Check if the user has any projects, if not then set view to "PROJECT_VIEW"
+        // to create a new one
+        let projectList = stores.ProjectStore.getAll();
+
+        let project = this.state.project;
+        if (!project && projectList) {
+            project = projectList.first();
+        }
+
+
+        let providerList;
+        providerList = stores.ProviderStore.getAll();
+
+
+        let provider = this.state.provider;
+        if (providerList) {
+            provider = provider || providerList.shuffle()[0];
```

Comment:
@lenards Makes sense! Thanks for clarifying, maybe it deserves a comment, since that wouldn't be clear looking at the code?